lua/nvim_lsp/rust_analyzer.lua

```diff
@@ -1,11 +1,88 @@
 local configs = require 'nvim_lsp/configs'
 local util = require 'nvim_lsp/util'
+local server_name = 'rust_analyzer'
+local release_url = 'https://api.github.com/repos/rust-analyzer/rust-analyzer/releases/latest'
+local postfix = '-linux'
+local bin = 'rust-analyzer'
+
+if vim.fn.has('mac') == 1 then
+  postfix = '-mac'
+elseif vim.fn.has('win32') == 1  or vim.fn.has('win64') == 1 then
+  postfix = '-windows.exe'
+end
+
+local function make_installer()
+  local I = {}
+  local P = util.path.join
+  local install_dir = P{util.base_install_dir, server_name}
+	local cmd_path = P{install_dir, bin}
+
+  local function getReleaseJson()
+    local json_string = vim.fn.system(string.format('curl -s "%s"', release_url), install_dir)
+    return vim.fn.json_decode(json_string)
+  end
+
+  function I.install()
+    local install_info = I.info()
+    vim.fn.mkdir(install_info.install_dir, "p")
+    local json = getReleaseJson()
+
+    local download_url = ''
+    local tag_name = json.tag_name
+    for _, value in pairs(json.assets) do
+      if (value ~= nil and string.match(value.browser_download_url, postfix..'$')) then
+        download_url = value.browser_download_url
+      end
+    end
+    if (download_url == '') then
+      print('Could not find download url. Aborting.')
+      return
+    end
+    util.sh(
+      string.format(
+        'echo "Starting download: %s" && ' ..
+        'curl -L "%s" > rust-analyzer && ' ..
+        'chmod 755 rust-analyzer && ' ..
+        'echo "%s" > .rust-analyzer-tag',
+      download_url, download_url, tag_name), install_info.install_dir)
+  end
+
+  function I.info()
+        return {
+            is_installed = util.path.exists(cmd_path);
+            install_dir = install_dir;
+            cmd = { cmd_path };
+        }
+  end
+
+  function I.configure(_)
+    local install_info = I.info()
+    local tag = vim.fn.system('cat .rust-analyzer-tag', install_info.install_dir)
+    local json = getReleaseJson()
+    if (json.tag_name == tag) then
+      return
+    end
+    local options = {"There is a new version of rust-analyzer. Would you like to update?", "1. Update", "2. Cancel"}
```

Comment:
Since I manage the installation of rust-analyzer myself, I assume I will get nagged about this every time I start Neovim? Obviously I could just change the config so it doesn't call `installer.info()`, but still... I'd rather not have to.

And what happens if I don't have internet access for some reason, will it barf errors?

Reply:
This is a really good point. Do you have a recommended way to resolve this? If a user points to a non-default  rust-analyzer command we can disable the auto-updating? Maybe include a specific flag to disable auto-updating?

I did check without internet and that seemed fine.

Reply:
> Do you have a recommended way to resolve this?

I'm afraid not.

> If a user points to a non-default rust-analyzer command we can disable the auto-updating? Maybe include a specific flag to disable auto-updating?

I suppose either of those will work. But I think the latter will be less error prone since it doesn't have to figure oute what a "non-default" rust-analyzer command is. The question is if it should default to on or off?

> I did check without internet and that seemed fine.

👍